scss/components/_badge.scss

```diff
@@ -18,6 +18,10 @@ $badge-color: $white !default;
 /// @type Color
 $badge-color-alt: $black !default;
 
+/// Coloring classes. The names in this list output classes in your CSS, like `.secondary`, `.success`, and so on. Each value must also be in the `$foundation-palette` map.
+/// @type List
+$badge-classes: ( secondary success warning alert ) !default;
```

Comment:
Make a search of `primary` in `_build/assets/css/foundation.css`, we don't strip it out everywhere.

I was searching a case where `.primary` could be useful to _restore_ properties, like in https://github.com/zurb/foundation-sites/issues/9424. It is useful in your example only because you consider `.primary` as a non-default modifier (it is not based on the default button config) and want to _apply_ its properties...